src/3rdparty/walkontable/src/table.js

```diff
@@ -773,83 +698,87 @@ class Table {
+   * 0-based index of row header
+   * @param {Number} level
    */
-  getLastRenderedRow() {
-    return this.wot.wtViewport.rowsRenderCalculator.endRow;
+  isRowHeaderLevelRendered(level) {
+    const columnHeaders = this.wot.getSetting('rowHeaders');
+    const columnHeadersCount = columnHeaders.length;
+    return level > (columnHeadersCount - 1);
   }
 
-  /**
-   * @returns {Number} Returns source index of last visible row
-   */
-  getLastVisibleRow() {
-    return this.wot.wtViewport.rowsVisibleCalculator.endRow;
-  }
+  isRowBeforeRenderedRows(row) {
+    const first = this.getFirstRenderedRow();
+    if (row < 0) {
+      if (first === 0) {
+        // first rendered row is above 0, so we can't expect any headers to be rendered
+        return false;
+      }
 
-  /**
-   * @returns {Number} Returns source index of last rendered column
-   */
-  getLastRenderedColumn() {
-    return this.wot.wtViewport.columnsRenderCalculator.endColumn;
-  }
+      return true;
 
-  /**
-   * @returns {Number} Returns -1 if no column is visible, otherwise source index of the last visible column
-   */
-  getLastVisibleColumn() {
-    return this.wot.wtViewport.columnsVisibleCalculator.endColumn;
-  }
+    }
 
-  isRowBeforeRenderedRows(row) {
-    return this.rowFilter && (this.rowFilter.sourceToRendered(row) < 0 && row >= 0);
+    if (first === -1) {
+      return true;
+    }
+    return row < first;
   }
 
   isRowAfterViewport(row) {
     return this.rowFilter && (row > this.getLastVisibleRow());
   }
 
   isRowAfterRenderedRows(row) {
-    if (Overlay.isOverlayTypeOf(this.wot.cloneOverlay, Overlay.CLONE_BOTTOM)
-      || Overlay.isOverlayTypeOf(this.wot.cloneOverlay, Overlay.CLONE_BOTTOM_LEFT_CORNER)) {
-      return false;
+    if (row < 0) {
+      const zeroBasedHeaderLevel = (-1 * row) - 1;
+      return this.isColumnHeaderLevelRendered(zeroBasedHeaderLevel);
     }
-
-    return this.rowFilter && (row > this.getLastRenderedRow());
+    return row > this.getLastRenderedRow();
   }
 
   isColumnBeforeViewport(column) {
     return this.columnFilter && (this.columnFilter.sourceToRendered(column) < 0 && column >= 0);
   }
 
   isColumnBeforeRenderedColumns(column) {
-    return this.columnFilter && (this.columnFilter.sourceColumnToVisibleRowHeadedColumn(column) < 0 && column >= 0);
+    const first = this.getFirstRenderedColumn();
+    if (column < 0) {
+      if (first === 0) {
+        // first rendered column is above 0, so we can't expect any headers to be rendered
+        return false;
+      }
+
+      return true;
+
+    }
+
+    if (first === -1) {
+      return true;
+    }
+    return column < first;
   }
 
   isColumnAfterViewport(column) {
     return this.columnFilter && (column > this.getLastVisibleColumn());
   }
 
   isColumnAfterRenderedColumns(column) {
+    if (column < 0) {
+      const zeroBasedHeaderLevel = (-1 * column) - 1;
```

Comment:
The same remark as for `getCell()`.

Reply:
Just to make sure, you mean this one: https://github.com/handsontable/handsontable/pull/6200#discussion_r319050906?

Reply:
Yes, exactly this one 😉